new(Aside): Add collapsible tab.

stefhatcher opened the pull request:
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

- adds a collapsible tab that toggles the aside open/closed
- supported for before and after, defaults to before (being on the right)
- tab is on the outside to not conflict with the scrollbar

<img width="1278" alt="Layouts : Layout - With Both Asides Collapsible And Main Padding ⋅ Storybook 2019-11-05 14-29-43" src="https://user-images.githubusercontent.com/306275/68251887-eb776500-ffd8-11e9-99a4-b437b3b5e3e2.png">
<img width="1278" alt="Layouts : Aside - With Collapsible Before ⋅ Storybook 2019-11-05 14-30-06" src="https://user-images.githubusercontent.com/306275/68251888-eb776500-ffd8-11e9-99d1-1352748d3071.png">
<img width="1278" alt="Layouts : Aside - With Collapsible After ⋅ Storybook 2019-11-05 14-30-22" src="https://user-images.githubusercontent.com/306275/68251889-ec0ffb80-ffd8-11e9-8136-e97af545add8.png">


## Motivation and Context

adding support to toggle aside open/closed

## Testing

added new stories to storybook

## Screenshots

original designs included the tab on the inside, which doesn't work well for scrollable content:
![Screen Shot 2019-11-05 at 11 01 00 AM](https://user-images.githubusercontent.com/306275/68241483-9bda6e80-ffc3-11e9-8ade-a7dcb0ca6a86.png)

with `scrollable` and `collapsible`:

<img width="1278" alt="Layouts : Aside - As A Scrollable Container With Collapsible Tab ⋅ Storybook 2019-11-05 14-30-38" src="https://user-images.githubusercontent.com/306275/68251861-e2869380-ffd8-11e9-8dd8-5c60fd2b43b2.png">


## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

milesj commented on packages/layouts/src/components/Aside/private/Tab.tsx:
```diff
@@ -0,0 +1,100 @@
+  });
+
+  let Icon = expanded ? (
+    <IconChevronLeft accessibilityLabel={labelCollapse} size={ICON_SIZE} />
```
`ExpandableIcon`?

milesj replied:
Need to handle LTR also.

stefhatcher replied:
doh. forgot about `ExpandableIcon`. will update!

milesj commented:
How does this work in the context of a `Layout` component?

stefhatcher commented:
@milesj updated with `DirectionalIcon` instead. also, for now, removed the animations. added a story to `Layout` to demonstrate it in a wider context!

milesj commented on packages/layouts/src/components/Aside/index.tsx:
```diff
@@ -39,24 +59,66 @@ export type Props = {
+
+  const handleCollapseToggle: React.DOMAttributes<HTMLButtonElement>['onClick'] = collapsible
+    ? () => {
+        setExpanded(!expanded);
```
`setExpanded(prev => !prev)`
